src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tailNode = tail;
+        Node<T> nodeToAdd = new Node<>(value, tail, null);
+        tail = nodeToAdd;
+        if (tailNode == null) {
+            head = nodeToAdd;
+        } else {
+            tailNode.next = nodeToAdd;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAddOperation(index);
+        Node<T> foundNode = findNodeByIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
let's call return here and unwrap the rest of the code, returning early is generally [a better practice](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)